### PR TITLE
Implement purge_lazy/_force() API of the Coarse provider

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Implement `purge_lazy/_force()` API of the Coarse provider.
Tests for the new API are included in https://github.com/oneapi-src/unified-memory-framework/pull/815.

Requires:
- [x] #800 
- [x] #803 
- [x] #806
- [x] #808 
- [x] #815 

### Checklist
- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

**Comment by ldorau** [comment]
A test added: `test/pools/jemalloc_pool_coarse_file.cpp`

**Comment by ldorau** [comment]
A test added: test/pools/pool_coarse_file.cpp

**Comment by ldorau** [comment]
A test added: `test/pools/pool_coarse_file.cpp`

**Comment by ldorau** [comment]
Tests for the new API are included in https://github.com/oneapi-src/unified-memory-framework/pull/815.

**Comment by ldorau** [comment]
Overall coverage rate:
lines......: 80.3% (4586 of 5708 lines)
https://github.com/oneapi-src/unified-memory-framework/actions/runs/11446915065

**Comment by ldorau** [comment]
PR is ready for review.

**Comment by ldorau** [comment]
Tests for the new API were added in https://github.com/oneapi-src/unified-memory-framework/pull/815.

**Comment by ldorau** [comment]
More tests added

**Comment by ldorau** [comment]
@bratpiorka PR is ready for review, tests added.

**Review by bratpiorka — APPROVED** [review]

**Review by PatKamin — APPROVED** [review]

File: test/disjointCoarseMallocPool.cpp [hunk 1]
```diff
@@ -1004,3 +1004,126 @@ TEST_P(CoarseWithMemoryStrategyTest,
...
+       disjointCoarseMallocPool_purge_no_upstream) {
+    umf_result_t umf_result;
+
+    const size_t init_buffer_size = 20 * MB;
+
+    // Preallocate some memory
+    std::unique_ptr<char[]> buffer(new char[init_buffer_size]);
+    void *buf = buffer.get();
+    ASSERT_NE(buf, nullptr);
+    memset(buf, 0, init_buffer_size);
```
[the remark this answers is not in the document]
Comment (ldorau) [reply test/disjointCoarseMallocPool.cpp:1018]: Done: https://github.com/oneapi-src/unified-memory-framework/pull/828

**Review by lukaszstolarczuk — APPROVED** [review]

**Pull request merged**